sql/plan/delete.go

```diff
@@ -95,6 +95,11 @@ func (p *DeleteFrom) Database() string {
 
 // RowIter implements the Node interface.
 func (p *DeleteFrom) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	// If an empty table is passed in (potentially from a bad filter) return an empty row iter.
```

Comment:
A cleaner solution would be to have emptyTable implement the deletable interface so we don't need to special case it here